data_steward/cdr_cleaner/cleaning_rules/clean_smoking_ppi.py

```diff
@@ -0,0 +1,362 @@
+                raise exc
+            else:
+                LOGGER.info(f'Loading of `{table_id}` completed.')
+
+    def get_query_specs(self, *args, **keyword_args) -> query_spec_list:
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+        queries_list = []
+
+        sandbox_query = dict()
+        sandbox_query[cdr_consts.QUERY] = SANDBOX_CREATE_QUERY.render(
+            project_id=self.project_id,
+            sandbox_dataset_id=self.sandbox_dataset_id,
+            new_smoking_rows=NEW_SMOKING_ROWS,
+            smoking_lookup_table=SMOKING_LOOKUP_TABLE,
+            dataset_id=self.dataset_id)
+        queries_list.append(sandbox_query)
+
+        delete_query = dict()
+        delete_query[cdr_consts.QUERY] = DELETE_INCORRECT_RECORDS.render(
+            project_id=self.project_id,
+            dataset_id=self.dataset_id,
+            sandbox_dataset_id=self.sandbox_dataset_id,
+            smoking_lookup_table=SMOKING_LOOKUP_TABLE)
+        queries_list.append(delete_query)
+
+        insert_query = dict()
+        insert_query[cdr_consts.QUERY] = INSERT_CORRECTED_RECORDS.render(
+            project_id=self.project_id,
+            dataset_id=self.dataset_id,
+            sandbox_dataset_id=self.sandbox_dataset_id,
+            new_smoking_rows=NEW_SMOKING_ROWS)
+        queries_list.append(insert_query)
+
+        return queries_list
+
+    def setup_validation(self, client):
+        """
+        Run required steps for validation setup
+        """
+        self.init_counts = self._get_counts(client)
+
+        if self.init_counts.get('total_rows') == 0:
+            raise RuntimeError('NO DATA EXISTS IN OBSERVATION TABLE')
+
+    def validate_rule(self, client):
+        """
+        Validates the cleaning rule which deletes or updates the data from the tables
+        """
+        clean_counts = self._get_counts(client)
+
+        if clean_counts.get('total_rows') == 0:
+            raise RuntimeError('LOST ALL DATA IN OBSERVATION TABLE')
+
+        if self.init_counts.get('total_rows') != clean_counts.get('total_rows'):
+            raise RuntimeError(
+                f'{self.__class__.__name__} unexpectedly changed table counts.\n'
+                f'initial count is: {self.init_counts.get("total_rows")}\n'
+                f'clean count is: {clean_counts.get("total_rows")}')
+
+        if clean_counts.get('child_count') or clean_counts.get('parent_count'):
+            raise RuntimeError(
+                f'{self.__class__.__name__} did not clean as expected.\n'
+                f'Found data in: {clean_counts}')
+
+    def _get_counts(self, client):
+        """
+        Counts query.
+        Used for job validation.
```

Comment:
Add a blank line after the initial summary line.  This is useful for automatic document generation, if we ever start using it. 

https://peps.python.org/pep-0257/#multi-line-docstrings